controller/server/gui.ml

```diff
@@ -257,29 +257,28 @@ module NetworkGui = struct
       let password_input =
         form_data |> List.assoc "proxy_password" |> List.hd |> non_empty
       in
-      match host_input, port_input, user_input, password_input with
-      (* Complete configuration was submitted *)
-      | Some host, Some port, Some user, Some password ->
-        return (Some (Service.Proxy.make ~user:user ~password:password host port))
-      (* Configuration was submitted without password, check whether it is
-         safe to refill from stored value *)
-      | Some host, Some port, Some user, None ->
-        (match current_proxy_opt with
-          | Some ({ credentials = Some { password } } as current_proxy) ->
-              let restored_proxy = Service.Proxy.make ~user:user ~password:password host port in
-              if current_proxy = restored_proxy then
-                (* Proxy configuration has not been touched *)
-                return (Some current_proxy)
-              else
-                (* Proxy configuration has been touched, it's unsafe to restore password *)
-                return (Some (Service.Proxy.make ~user:user ~password:"" host port))
-          | _ ->
-              (* No existing proxy credentials, use empty password *)
-              return (Some (Service.Proxy.make ~user:user ~password:"" host port))
-        )
+      let keep_password = 
+        form_data |> List.assoc_opt "keep_password" |> Option.is_some
+      in
+      let password = 
+        match (keep_password, current_proxy_opt) with
+        | (true, Some ({ credentials = Some { password } })) -> Ok (Some password)
```

Comment:
This seems to introduce a vulnerability. It allows to change server while keeping the password unchanged, which allows revealing the password to a server an attacker controls.

Reply:
Good point.

We could check in the backend if the host (and port ?) didn’t change. 

Ideally, this should be reflected in the UI as well, but we may need a web component to generate the complete section, because there are interactions between elements: when changing the host, remove “keep password” and show a blank password input.

Maybe we should just remove this logic and let the user type the password if a modification is made ?

Reply:
> Maybe we should just remove this logic and let the user type the password if a modification is made ?

I forgot a bit how the previous mechanism worked, and may have misunderstood it a bit.

Password was kept only if proxy configuration was the same. This is useful in case someone wants to modify only the static IP configuration. Typing the password in the proxy form is not appropriate when someone wants to modify only the static IP configuration.

So, I think the previous solution was quite good, and only missed removing the password: add a “remove password” button to take care of this ?